community/codegen/src/main/java/org/neo4j/codegen/source/Configuration.java

```diff
@@ -98,49 +108,18 @@ public boolean isSet( SourceCode flag )
         return flags != null && flags.contains( flag );
     }
 
-    public void addSourceVisitor( SourceVisitor visitor )
-    {
-        if ( sourceVisitors == null )
-        {
-            sourceVisitors = new ArrayList<>();
-        }
-        sourceVisitors.add( visitor );
-    }
-
     public void visit( TypeReference reference, StringBuilder source )
     {
-        if ( sourceVisitors != null )
+        for ( SourceVisitor visitor : sourceVisitors )
         {
-            for ( SourceVisitor visitor : sourceVisitors )
-            {
-                visitor.visitSource( reference, source );
-            }
+            visitor.visitSource( reference, source );
         }
     }
 
-    public void addWarningsHandler( WarningsHandler handler )
-    {
-        if ( warningsHandlers == null )
-        {
-            warningsHandlers = new ArrayList<>();
-        }
-        warningsHandlers.add( handler );
-    }
-
     public WarningsHandler warningsHandler()
     {
-        if ( warningsHandlers == null )
-        {
-            return WarningsHandler.NO_WARNINGS_HANDLER;
-        }
-        if ( warningsHandlers.size() == 1 )
-        {
-            return warningsHandlers.get( 0 );
```

Comment:
I like not having to create the additional object indirection if it isn't required.